src/main/kotlin/org/rust/lang/core/resolve2/DefCollector.kt

```diff
@@ -310,7 +313,7 @@ class DefCollector(
         if (macros.isEmpty()) return
         val batches = macros.splitInBatches(100)
 
-        val result = if (pool != null) {
+        val result = if (pool != null && EXPAND_MACROS_IN_PARALLEL.asBoolean()) {
```

Comment:
Maybe it's better just to pass `null` as pool value during `DefMapsBuilder` creation?